src/sql/workbench/services/accountManagement/browser/accountDialog.ts

```diff
@@ -406,11 +425,21 @@ export class AccountDialog extends Modal {
 
 		// Remove the list view from the split view
 		this._splitView!.removeView(providerView.view.index!);
+		// Decrement index for each provider
+		this._providerViewsMap.forEach((provider) => {
+			if (provider.view.index > providerView.view.index) {
+				provider.view.index--;
+			}
+		})
 		this._splitView!.layout(DOM.getContentHeight(this._container!));
 
 		// Remove the list view from our internal map
 		this._providerViewsMap.delete(removedProvider.id);
 		this.logService.debug(`Provider ${removedProvider.id} removed`);
+		// Update view after removing provider
+		if (this._splitViewContainer!.hidden || this._providerViewsMap.size > 1) {
+			this.showSplitView();
+		}
 		this.layout();
```

Comment:
Before layout, please also updateAccounts as done when adding a provider.

Reply:
fixed the problem this comment was referring to, it was a different issue